flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/util/RecordEmitter.java

```diff
@@ -205,6 +205,10 @@ protected void emitRecords() {
 						}
 					}
 				}
+				if (!running) {
```

Comment:
This wouldn't be necessary if the thread was interrupted when the fetcher terminates. Once we confirm that this change has the intended effect, let's merge it and cherry-pick into the other release branches. I will take a look at improving the thread management in the fetcher as a follow-up.